crates/flowistry_ide/src/mutations/find_mutations.rs

```diff
@@ -0,0 +1,52 @@
+use flowistry::{
+  indexed::IndexSetIteratorExt,
+  infoflow::mutation::ModularMutationVisitor,
+  mir::{aliases::Aliases, utils::PlaceExt},
+};
+use log::debug;
+use rustc_hir::def_id::DefId;
+use rustc_middle::{
+  mir::{visit::Visitor, Body, Location, Place},
+  ty::TyCtxt,
+};
+
+pub fn find_mutations(
+  tcx: TyCtxt<'tcx>,
+  body: &Body<'tcx>,
+  def_id: DefId,
+  place: Place<'tcx>,
+  aliases: Aliases<'tcx>,
+) -> Vec<Location> {
+  let mut locations = vec![];
+  let pointer_aliases = place
```

Comment:
Two things:
1. As a style thing, if your method chain is over 4 lines of code, I would split it up with a good variable name for the intermediates. E.g. `let foo = (...).flatten(); let pointer_aliases = foo.map(...);`
2. We use this code also in the slicing feature, so I would turn it into a standalone function on the `Aliases` struct. Something like, `aliases.reachable_values(place)` since the goal is to resolve all the pointers reachable from a given place to concrete locations.